apps/meteor/app/lib/server/functions/getRoomByNameOrIdWithOptionToJoin.ts

```diff
@@ -88,7 +88,7 @@ export const getRoomByNameOrIdWithOptionToJoin = async ({
 	// If the room type is channel and joinChannel has been passed, try to join them
 	// if they can't join the room, this will error out!
 	if (room.t === 'c' && joinChannel) {
-		await addUserToRoom(room._id, user);
+		await Room.join({ room, user });
```

Comment:
maybe we should have a comment (jsdoc) on `addUserToRoom` mentioning that it doesn't check permissions, as it still validates a ton of other stuff.